explorer/websocket.go

```diff
@@ -164,3 +184,58 @@ func (wsh *WebsocketHub) run() {
 		}
 	}
 }
+
+// addTxToBuffer adds a tx to the buffer, then initiates a buffer send and resets
+// the send ticker if the buffer is full
+func (wsh *WebsocketHub) addTxToBuffer(tx *MempoolTx) {
+	wsh.RLock()
+	wsh.newTxBuffer = append(wsh.newTxBuffer, tx)
+	wsh.RUnlock()
+	if len(wsh.newTxBuffer) == newTxBufferSize {
+		go wsh.sendNewTxs()
+		wsh.bufferTickerChan <- tickerSigReset
+	}
+}
+
+// sendNewTxs sends all txs in the buffer if any and clears the buffer
+func (wsh *WebsocketHub) sendNewTxs() {
+	wsh.RLock()
+	if len(wsh.newTxBuffer) == 0 {
+		wsh.RUnlock()
+		return
+	}
+	txs := make([]*MempoolTx, len(wsh.newTxBuffer))
+	copy(txs, wsh.newTxBuffer)
+	wsh.newTxBuffer = make([]*MempoolTx, 0, newTxBufferSize)
+	wsh.RUnlock()
+	log.Debugf("Signaling %d new tx to %d clients", len(txs), len(wsh.clients))
+	for signal, client := range wsh.clients {
+		client.RLock()
+		client.newTxs = txs
```

Comment:
All clients get the same slice of transactions, so i suggest a different approach:  use the `WebsocketHub`'s `run()` loop to handle a lot of these concurrency issues.  That is, follow the pattern like `RegisterClient`/`registerClient` where the former sends on a channel only read by `run`, and the latter is called when `run` gets to it.

The main problem that needs to be addresses with this PR is that `(wsh *WebsocketHub) sendNewTxs()`, which ranges over `wsh.clients`, is launched as a goroutine and runs concurrently with the main `run` loop, which registers and unregisters clients in `wsh.clients`.

I'm open to any solution, it doesn't need to be the all-event-loop idea I mentioned here, but concurrent map access (clients) needs to be addressed.